src/components/ec-datepicker/ec-datepicker.spec.js

```diff
@@ -0,0 +1,387 @@
+
+      await inputWrapper.setProps({
+        areWeekendsDisabled: false,
+      });
+
+      inputWrapper.vm.$nextTick();
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-19').classes('flatpickr-disabled')).toBeFalsy();
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-20').classes('flatpickr-disabled')).toBeFalsy();
+    });
+
+    it('should update the disabled date', async () => {
+      const { inputWrapper, calendarWrapper } = mountDatepicker({
+        disabledDates: {
+          '2022-02-22': 'Bank holiday',
+        },
+      });
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-22').classes('flatpickr-disabled')).toBeTruthy();
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-22').attributes('title')).toBe('Bank holiday');
+
+      await inputWrapper.setProps({
+        disabledDates: {
+          '2022-02-23': 'Bank holiday',
+        },
+      });
+
+      inputWrapper.vm.$nextTick();
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-22').classes('flatpickr-disabled')).toBeFalsy();
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-22').attributes('title')).toBe(undefined);
+
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-23').classes('flatpickr-disabled')).toBeTruthy();
+      expect(calendarWrapper.findByDataTest('ec-datepicker__calendar-day--2022-02-23').attributes('title')).toBe('Bank holiday');
+    });
+  });
+
+  describe('v-model', () => {
+    it('should have preselected default date', () => {
+      const { inputWrapper } = mountDatepickerAsTemplate(
+        '<ec-datepicker v-model="model" />',
+        {},
+        {
+          data() {
+            return {
+              model: '2022-02-22',
+            };
+          },
+        },
+      );
+
+      expect(inputWrapper.vm.model).toBe('2022-02-22');
+    });
+
+    it('should update the value of the calendar when I type a value', () => {
+      const { inputWrapper } = mountDatepickerAsTemplate(
+        '<ec-datepicker v-model="model" />',
+        {},
+        {
+          data() {
+            return {
+              model: '',
+            };
+          },
+        },
+      );
+
+      inputWrapper.findByDataTest('ec-input-field__input').setValue('2022-02-23');
+
+      expect(inputWrapper.vm.model).toBe('2022-02-23');
+    });
+
+    it('should update the value of the calendar when I select a value from the datepicker', () => {
+      const { inputWrapper, calendarWrapper } = mountDatepickerAsTemplate(
+        '<ec-datepicker v-model="model" />',
+        {},
+        {
+          data() {
+            return {
+              model: '',
+            };
+          },
+        },
+      );
+
+      calendarWrapper
```

Comment:
click() should be awaited.

Reply:
As a general rule when we click to await or only now because potentially flatpickr can take some time to calculate? Shall we do the same for example when we check the click action that toggles the visibility of a <div> in one of our own components?